From 3c1a0119560999e0ad8946bc2411d7215caf3f10 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:05:27 -0400
Subject: [powerpc] eeh: Reorder output messages

Message-id: <1410545655-205645-99-git-send-email-dzickus@redhat.com>
Patchwork-id: 94535
O-Subject: [RHEL7 PATCH 098/626] powerpc/eeh: Reorder output messages
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 5293bf97a27e1be8ac6096aa198ff6a9e3e6837c
Author: Gavin Shan <shangw@linux.vnet.ibm.com>
Date:   Fri Sep 6 09:00:05 2013 +0800

    powerpc/eeh: Reorder output messages

    We already had some output messages from EEH core. Occasionally,
    we can see the output messages from EEH core before the stack
    dump. That's not what we expected. The patch fixes that and shows
    the stack dump prior to output messages from EEH core.

    Signed-off-by: Gavin Shan <shangw@linux.vnet.ibm.com>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/kernel/eeh.c b/arch/powerpc/kernel/eeh.c
index 40829aa..3759df4 100644
--- a/arch/powerpc/kernel/eeh.c
+++ b/arch/powerpc/kernel/eeh.c
@@ -327,11 +327,11 @@ static int eeh_phb_check_failure(struct eeh_pe *pe)
  /* Isolate the PHB and send event */
  eeh_pe_state_mark(phb_pe, EEH_PE_ISOLATED);
  eeh_serialize_unlock(flags);
- eeh_send_failure_event(phb_pe);
 
  pr_err("EEH: PHB#%x failure detected\n",
   phb_pe->phb->global_number);
  dump_stack();
+ eeh_send_failure_event(phb_pe);
 
  return 1;
 out:
@@ -454,8 +454,6 @@ int eeh_dev_check_failure(struct eeh_dev *edev)
  eeh_pe_state_mark(pe, EEH_PE_ISOLATED);
  eeh_serialize_unlock(flags);
 
- eeh_send_failure_event(pe);
-
  /* Most EEH events are due to device driver bugs.  Having
   * a stack trace will help the device-driver authors figure
   * out what happened.  So print that out.
@@ -464,6 +462,8 @@ int eeh_dev_check_failure(struct eeh_dev *edev)
   pe->addr, pe->phb->global_number);
  dump_stack();
 
+ eeh_send_failure_event(pe);
+
  return 1;
 
 dn_unlock:
-- 
1.7.1